### ISSUE TITLE
Compilation error in 1.18.0 - no function clause matching in Exception.format_mfa/3

### ISSUE THREAD

**Issue by seantanly** [elixir-lang/elixir#14105]
### Elixir and Erlang/OTP versions

Erlang/OTP 27 [erts-15.2] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit]

Elixir 1.18.0 (compiled with Erlang/OTP 27)

### Operating system

MacOS 15.2

### Current behavior

I encountered the following compilation error when running
```
mix clean && mix compile
```

The actual module/function names are obfuscated in the console output.
```
==> api
Compiling 384 files (.ex)
** (EXIT from #PID<0.94.0>) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Exception.format_mfa/3
        (elixir 1.18.0) lib/exception.ex:725: Exception.format_mfa({:XXXX, [version: 7, line: 161, column: 16], nil}, :build_YYYY 1)
        (elixir 1.18.0) lib/module/types/apply.ex:798: Module.Types.Apply.format_diagnostic/1
        (elixir 1.18.0) lib/module/parallel_checker.ex:299: anonymous fn/2 in Module.ParallelChecker.group_warnings/1
        (elixir 1.18.0) lib/enum.ex:2546: Enum."-reduce/3-lists^foldl/2-0-"/3
        (elixir 1.18.0) lib/module/parallel_checker.ex:298: Module.ParallelChecker.group_warnings/1
        (elixir 1.18.0) lib/module/parallel_checker.ex:241: Module.ParallelChecker.check_module/3
        (elixir 1.18.0) lib/module/parallel_checker.ex:90: anonymous fn/7 in Module.ParallelChecker.inner_spawn/6
```

Looks like the error occurs while the compiler is trying to generate and print type errors.

### Expected behavior

The compilation warning/error should be printed in the console instead of the compilation process exiting.

**Comment by josevalim** [comment]
Do you have a source code that could reproduce this?

Perhaps share the function at line  `Exception.format_mfa({:XXXX, [version: 7, line: 161, column: 16], nil}, :build_YYYY 1)`? :)(

**Comment by josevalim** [comment]
Nevermind, I was able to reverse engineer it :D

**Comment by josevalim** [comment]
Fixed on main and v1.18, thank you.
